src/main/kotlin/org/rust/cargo/runconfig/RustBacktraceFilter.kt

```diff
@@ -1,17 +1,137 @@
 package org.rust.cargo.runconfig
 
+import com.intellij.execution.filters.Filter
+import com.intellij.execution.filters.OpenFileHyperlinkInfo
 import com.intellij.execution.filters.RegexpFilter
+import com.intellij.openapi.editor.colors.EditorColorsManager
+import com.intellij.openapi.editor.colors.TextAttributesKey
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.psi.PsiDocumentManager
+import org.rust.cargo.project.CargoProjectDescription
+import org.rust.cargo.util.cargoProject
+import org.rust.cargo.util.getPsiFor
+import org.rust.cargo.util.modulesWithCargoProject
+import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustNamedElement
+import org.rust.lang.core.resolve.RustResolveEngine
+import org.rust.lang.core.symbols.RustPath
+import org.rust.lang.core.symbols.RustPathHead
+import org.rust.lang.core.symbols.RustPathSegment
+import java.util.*
+import java.util.regex.Pattern
 
 /**
- * Detects messages about panics and adds source code links to them.
+ * Adds features to stack backtraces:
+ * - Wrap function calls into hyperlinks to source code.
+ * - Turn source code links into hyperlinks.
+ * - Dims funcion hashcodes to remove noise.
  */
 class RustBacktraceFilter(
     project: Project,
     cargoProjectDir: VirtualFile
-) : RegexpFileLinkFilter(
-    project,
-    cargoProjectDir,
-    "^\\s+at ${RegexpFilter.FILE_PATH_MACROS}:${RegexpFilter.LINE_MACROS}$") {
+) : Filter {
+    private val sourceLinkFilter = RegexpFileLinkFilter(project, cargoProjectDir, "^\\s+at ${RegexpFilter.FILE_PATH_MACROS}:${RegexpFilter.LINE_MACROS}$")
+    private val backtraceItemFilter = RustBacktraceItemFilter(project)
+
+    override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
+        backtraceItemFilter.applyFilter(line, entireLength)?.let { return it }
+        sourceLinkFilter.applyFilter(line, entireLength)?.let { return it }
+        return null
+    }
+}
+
+/**
+ * Adds hyperlinks to function names in backtraces
+ */
+private class RustBacktraceItemFilter(
+    val project: Project
+) : Filter {
+    private val pattern = Pattern.compile("^(\\s*\\d+:\\s+0x[a-f0-9]+ - )(.+)(::h[0-9a-f]+)$")!!
+    private val docManager = PsiDocumentManager.getInstance(project)
+
+    override fun applyFilter(line: String, entireLength: Int): Filter.Result? {
+        val matcher = pattern.matcher(line)
+        if (!matcher.find()) return null
+        val funcName = matcher.group(2)
```

Comment:
Small stylistic not, but what do you think about binding all groups in the same place, like

```
val funcPath = matcher.group(1)
val funcName = matcher.group(2)
val funcHash = matcher.group(3)
```